src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -3188,6 +3188,16 @@ private bool isFieldStatic(CORINFO_FIELD_STRUCT_* fldHnd)
             return HandleToObject(fldHnd).IsStatic;
         }
 
+        private bool canOmitPinning(CORINFO_FIELD_STRUCT_* fldHnd)
+        {
+            FieldDesc field = HandleToObject(fldHnd);
+            if (!field.IsStatic || field.IsThreadStatic || field.HasGCStaticBase || field.OwningType.IsCanonicalSubtype(CanonicalFormKind.Any))
```

Comment:
HasGCStaticBase condition should not be needed for NativeAOT.

See https://github.com/dotnet/runtimelab/pull/870